state/factory/workingset_test.go

```diff
@@ -0,0 +1,161 @@
+package factory
+
+import (
+	"context"
+	"errors"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/pkg/unit"
+	"github.com/iotexproject/iotex-core/test/identityset"
+	"github.com/iotexproject/iotex-core/testutil"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
+	"time"
+	//"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	//"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+func TestWriteReadView(t *testing.T) {
+	require := require.New(t)
+	sf, err := NewFactory(config.Default, InMemTrieOption())
+	require.NoError(err)
+	ctx := protocol.WithBlockchainCtx(
+		protocol.WithRegistry(context.Background(), protocol.NewRegistry()),
+		protocol.BlockchainCtx{Genesis: config.Default.Genesis},
+	)
+	require.NoError(sf.Start(ctx))
+	ws, err := sf.(workingSetCreator).newWorkingSet(ctx, 1)
+
+	t.Run("ReadView call error", func(t *testing.T) {
+		retrdf, retErr := ws.ReadView("")
+		expectedErrors := errors.New("name : name does not exist")
+		require.EqualError(retErr, expectedErrors.Error())
```

Comment:
require.Equal(dock.ErrNoName, errors.Cause(retErr))
no need to define a new expectedErrors